docs/processors/standalone/building.mdx

```diff
@@ -0,0 +1,141 @@
+---
+title: 'Build your own'
+sidebar_position: 0
```

Comment:
Sorry for this long comment in advance, and I should have probably posted it sooner, but the more I think about it, the more I see the need to restructure this page entirely. I'm posting this comment on this line, simply so that we can have a discussion.

My proposal is to start from scratch and not follow the readme we had in the SDK. I see this as more of an in-depth page that explains in detail what to do, while the readme previously only tried to give a quick overview of the most important parts. I sketched out the structure I imagined:

---

# Build your own

Explain that you can build your own processor using the SDK, that we currently provide a Go SDK, that you may poke us in an issue if you want a different language, that you can look at the Wasm page if you want to build your own.

Mention that the SDK exposes two ways of building a processor, one for very simple ones, the other that is a bit more complicated, but gives you more control.

## Using `sdk.NewProcesorFunc`

Explain `sdk.NewProcessorFunc` approach. Explain when to use it, what are its limitations and when to switch to the full processor approach.

## Using `sdk.Processor`

Explains how to build the full-blown processor. I'd try to break this down into the parts that need to be implemented.

### Specifications

Explain what are specifications, what are they good for. Definitely mention `paramgen` and explain how to use it to generate the parameter specifications.

### Configure

Explain what to implement (and what _not_ to) in this function. Definitely mention the utilities we provide and how most configuration implementations will look like. Mention how to add custom validations here.

### Open

Explain what is supposed to happen here (e.g. opening connections, initializing resources). Mention this one is optional.

### Process

Here we should go into detail on what to take into account when implementing this.

### Teardown

Explain that this is the counterpart to `Open`, that it's meant to close any open connections and resources. This one is optional, same as `Open`.

## Logging

Explain that the SDK provides a logger, that it should be retrieved using the context provided by Conduit, that log level should be "trace" in the hot path.

## Compiling the processor

Mention the env variables that need to be set and why that's the case.

At the end point to the main "Standalone processors" page to see how the compiled processor can then be used in a pipeline.

---

Let me know what you think. cc @raulb @hariso

Reply:
@lovromazgon I agree with that suggestion. I see the value of presenting different sets of information between the repository readme and this page. I think we should also go in the direction of showing code snippets of more in depth language instructions on each GitHub repository, and leaving these documents to express more context on what it is, and why it could be useful.

I like the structure of what you're presenting because it reminds me of what we have in other pages recently added / updated which can give you a quick overview of what's possible (or not -suggesting to create an issue-), and then exposing the different options and _components_ with a different language.